### DISCUSSION TITLE
Item link in Search Results returns list

### DISCUSSION THREAD

**Discussion by blopez-co-slo-ca-us** [microsoft-search/pnp-modern-search#1659] [Q&A]
Hello All, just started toying with modern search so please be patient..

Using the modern search web parts (V4) to search across three repositories related to my hub: 
**Documents, Lists and Site Pages**
Web Part: **PnP - Search Results**  
Data sources: **SharePoint Search**
Available layouts:**Details**

This mostly works fine. I get a result set that is a mix of documents, list items and site pages.  When I click a result for a document or
site page I am taken to that doc or site page.  When I click a result for a list item I am taken to the top level of the list
and not the list item.

Under Layout Options -> Manage columns
The value for the column Title:
`<a href="{{slot item @root.slots.PreviewUrl}}" target="_blank" style="color: {{@root.theme.semanticColors.link}}">
	{{slot item @root.slots.Title}}
</a> `

It seems _@root.slots.PreviewUrl_ is returning

> https://<tenant>.sharepoint.com/sites/My-Site/Lists/Web%20Links/108_.000?web=1

However, for list items this might be better:

> https://<tenant>.sharepoint.com/sites/My-Site/Lists/Web%20Links/DispForm.aspx?ID=108

Can I modify the webhook code to change the returned URL based on the SPContentType?  Some kind of if/else maybe?  Am I taking the wrong approach? Any help would be appreciated.

**Comment by wobba** [comment]
You can add an if/else in your template. And I'll take a look if we can improve the default logic for list items. You want to use e.g. the `item.Path`property instead of slots, or use the slot containing `Path`.
